### PR TITLE
Turnanim

### PR DESCRIPTION
**Pull request by sunqirui1987**

### CODE REVIEW

File: test/MonkeyAndCrocodile/res/sprites/Crocodile/index.json [hunk 1]
```diff
@@ -13,17 +13,26 @@
   "currentCostumeIndex": 0,
   "animations": {
     "sink": {
-      "play": "chomp",
```
Comment (xushiwei) [inline test/MonkeyAndCrocodile/res/sprites/Crocodile/index.json:16]: Let's discuss it

File: sprite.go [hunk 2]
```diff
@@ -1062,3 +1112,16 @@ func (p *Sprite) ShowVar(name string) {
 }
 
 // -----------------------------------------------------------------------------
+
+//get sprite size
+func (p *Sprite) Width() int {
```
Comment (xushiwei) [inline sprite.go:1117]: merged. please remove it.

File: anim.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
+package spx
+
+import (
+	"github.com/goplus/spx/internal/anim"
+	"github.com/goplus/spx/internal/tools"
+)
+
+type ANIMSTATUS uint8
```
[the remark this answers is not in the document]
Comment (sunqirui1987) [reply anim.go:8]: ok,  hide it first, then open it when need it

File: anim.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+const (
+	AnimstatusPlaying ANIMSTATUS = iota
```
[the remark this answers is not in the document]
Comment (sunqirui1987) [reply anim.go:11]: up!

File: game.go [hunk 5]
```diff
@@ -723,6 +754,10 @@ func (p *Game) getGdiPos(x, y float64) (int, int) {
 	screenW, screenH := p.size()
 	return int(x) + (screenW >> 1), (screenH >> 1) - int(y)
 }
+func (p *Game) getPosFormGdi(x, y float64) (int, int) {
```
[the remark this answers is not in the document]
Comment (sunqirui1987) [reply game.go:757]: ok

File: anim.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+	keyframelist         []*anim.AnimationKeyFrame
+	animationStartedDate int
+
+	//playing
+	playingCallback func(int, interface{})
+	//stop
+	stopCallback func()
+}
+
+func NewAnim(name string, valtype ANIMVALTYPE, fps float64, totalframe int) *Anim {
```
[the remark this answers is not in the document]
Comment (sunqirui1987) [reply anim.go:40]: hide it first,

File: sprite.go [hunk 7]
```diff
@@ -34,7 +36,8 @@ const (
 type Sprite struct {
 	baseObj
 	eventSinks
-	g *Game
+	g     *Game
+	gamer reflect.Value
```
[the remark this answers is not in the document]
Comment (sunqirui1987) [reply sprite.go:40]: anim events need to visit "gamer"

File: test/MonkeyAndCrocodile/res/sprites/Crocodile/index.json (hunk 1 above)
Comment (sunqirui1987) [reply test/MonkeyAndCrocodile/res/sprites/Crocodile/index.json:16]: ok

**Pull request merged**